server/chat-commands/room-settings.ts

```diff
@@ -300,7 +301,12 @@ export const commands: ChatCommands = {
 			if (!Users.Auth.supportedRoomPermissions(room).includes(perm)) {
 				return this.errorReply(`${perm} is not a valid room permission.`);
 			}
-			if (!(room.auth.atLeast(user, '#') && Users.Auth.hasPermission(user, perm, null, room))) {
+			if (!room.auth.atLeast(user, '#') ||
+				(
+					[...ROOM_PERMISSIONS, ...GLOBAL_PERMISSIONS].includes(perm as GlobalPermission | RoomPermission) &&
+					Users.Auth.hasPermission(user, perm, null, room)
+				)
+			) {
 				return this.errorReply(`/permissions set - Access denied.`);
 			}
```

Comment:
These would almost be better as two separate tests.

```suggestion
			if (!room.auth.atLeast(user, '#')) {
				return this.errorReply(`/permissions set - You must be at least a Room Owner to set permissions.`);
			}
			if (
				Users.Auth.ROOM_PERMISSIONS.includes(perm as RoomPermission) &&
				Users.Auth.hasPermission(user, perm, null, room)
			) {
				return this.errorReply(`/permissions set - You can't set the permission "${perm}" because you already have it.`);
			}
```

Actually this test looks weird. What is this actually supposed to be guarding?

Reply:
It's supposed to be preventing users from setting permissions for things they can't do; for example, a Room Owner who is a Global Voice shouldn't be able to set the permissions for appointing new ROs to # and then demote/promote ROs.

Reply:
Shouldn't it be `!Users.Auth.hasPermission` rather than `Users.Auth.hasPermission`, then?

Reply:
```suggestion
			if (!room.auth.atLeast(user, '#')) {
				return this.errorReply(`/permissions set - You must be at least a Room Owner to set permissions.`);
			}
			if (
				Users.Auth.ROOM_PERMISSIONS.includes(perm as RoomPermission) &&
				!Users.Auth.hasPermission(user, perm, null, room)
			) {
				return this.errorReply(`/permissions set - You can't set the permission "${perm}" because you don't have it.`);
			}
```

Reply:
Just to make sure: Was the previous code just bugged?

Reply:
Yes, it prevented ROs from setting permissions for things like blacklisting.